src/Microsoft.AspNetCore.Http/HeaderDictionary.cs

```diff
@@ -125,25 +130,13 @@ public HeaderDictionary(int capacity)
         /// Gets the number of elements contained in the <see cref="HeaderDictionary" />;.
         /// </summary>
         /// <returns>The number of elements contained in the <see cref="HeaderDictionary" />.</returns>
-        public int Count
-        {
-            get
-            {
-                return Store?.Count ?? 0;
-            }
-        }
+        public int Count => Store?.Count ?? 0;
 
         /// <summary>
         /// Gets a value that indicates whether the <see cref="HeaderDictionary" /> is in read-only mode.
         /// </summary>
         /// <returns>true if the <see cref="HeaderDictionary" /> is in read-only mode; otherwise, false.</returns>
-        public bool IsReadOnly
-        {
-            get
-            {
-                return false;
-            }
-        }
+        public bool IsReadOnly { get; set; }
```

Comment:
Should this throw if it's been set to ReadOnly? It'd be weird if people reset this bit to work around the exception.

Reply:
I'm not as concerned about that, they'd have to downcast to get to it in the first place and at that point it becomes pretty obvious they're doing something wrong.